Bug 1729505: Fix hugepages node config

adellape opened the pull request:
xref: https://bugzilla.redhat.com/show_bug.cgi?id=1729505

Preview (internal): http://file.rdu.redhat.com/~adellape/072919/hugepages/scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.html#configuring-huge-pages_huge-pages

adellape commented:
@openshift/team-documentation 	PTAL.

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
+. Copy the default tuned profile and customize it to create a tuned profile
+allocating huge pages for your labeled node.
+
+.. Export the default tuned profile and save it to a file (`tuned.yaml`):
```
> Avoid using parentheses in running text because text in parentheses is decreased in importance. 

s/file (`tuned.yaml`)/file that is named `tuned.yaml`

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
-.. In the `profile` section, add the following tuned profile for allocating huge
-pages:
+.. To rename the profile to `hugepages`, edit the `tuned.yaml` file and update
+the `metadata.name` parameter:
```
s/parameter/parameter value

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
+  name: hugepages <1>
+  namespace: openshift-cluster-node-tuning-operator
 ----
+<1> Rename the profile.
```
s/Rename the profile./Set the `name` parameter value to `hugepages`.

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
-.. To apply the profile to the correct node, add the following to the `match`
+.. To add a tuned profile for allocating huge pages and associate the profile to
+the correct node, add the following `profile` and `match` sections to the `spec`
 section:
```
s/section/section of the `tuned.yaml` file

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
+    priority: 30
+    profile: node-hugepages
+----
+<1> Add a `profile` section.
```
Are there any customizations that the user should consider in these two sections, or must they use these exact settings?

adellape replied:
^ @skordas 	@jmencak 	?

jmencak replied:
The CR is a good start -- it works now...  I'd leave this as is.  As for customizations, perhaps the number of "persistent" huge pages in the kernel's huge page pool (nr_hugepages) and profile priority should be considered as "customizations".

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
+<1> Add a `profile` section.
+<2> Add a `match` section.
+
+. Create the new, custom  `hugepages` tuned profile using the modified
```
~new,~
s/using/by using
remove double space after "custom"

kalexand-rh commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
 
-The Operator applies the new profile to the correct node and allocates huge
-pages:
+After completing this procedure, the Operator applies the new profile to the
```
Is this the result of running the previous step? Is this output displayed after you run the `oc create` command? If so, I'd connect the remainder of the module to the previous step and ~After completing this procedure,~.

adellape replied:
No, I believe these are other logs. Would be good to show the command being run to view these logs here.

jmencak replied:
For that, you need to know which containerized tuned pod to query. You can do something like `oc logs <tuned_pod_on_node_using_hugepages> -n  openshift-cluster-node-tuning-operator | tail -n2` and with 4.2 you'll get
```
2019-08-08 07:20:41,276 INFO     tuned.plugins.plugin_sysctl: reapplying system sysctl
2019-08-08 07:20:41,286 INFO     tuned.daemon.daemon: static tuning from profile 'node-hugepages' applied
```

Note that in 4.2, containerized tuned no longer runs in one-shot mode, so you'll not get the ` tuned.daemon.daemon: terminating Tuned` line, so it should be dropped.  Personally, I'd say something like this in the docs:

```
$ oc logs <tuned_pod_on_node_using_hugepages> -n  openshift-cluster-node-tuning-operator | grep 'applied$' | tail -n1
2019-08-08 07:20:41,286 INFO     tuned.daemon.daemon: static tuning from profile 'node-hugepages' applied
```

Then the log output will be valid for 4.1 and 4.2

skordas commented on modules/configuring-huge-pages.adoc:
```diff
@@ -10,43 +14,66 @@ Node Tuning Operator to allocate huge pages on a specific node.
 +
 ----
-$ oc edit tuned
+$ oc get tuned --export -o yaml > tuned.yaml
```
Missing projects to get tuned.
Moreover I thing is good to change name of exported file to something like hugepages_tuning.yaml
`oc get tuned --export -o yaml -n openshift-cluster-node-tuning-operator > tuned.yaml`

skordas commented:
I think better is not to use copy of default tuned. It has more profiles (openshift, openshift-control-plane, openshift-node, openshift-control-plane-es, openshift-node-es)
So when we want to use copy of that file to create hugepages profile, user need to remove all other profiles from copy.

Better is to use sample provided by @jmencak in bugzilla.

jmencak commented:
> I think better is not to use copy of default tuned. It has more profiles (openshift, openshift-control-plane, openshift-node, openshift-control-plane-es, openshift-node-es)
> So when we want to use copy of that file to create hugepages profile, user need to remove all other profiles from copy.
> 
> Better is to use sample provided by @jmencak in bugzilla.

Exactly what I thought when I saw this PR.  I think it would be better not to copy the default tuned, but I'm probably biased.

adellape commented:
Thanks, updated per feedback.

Changed the procedure to use the sample from the BZ instead of exporting and modifying the default. PTAL again @skordas 	@jmencak. Thank you!

jmencak commented:
I'd only suggest showing only one log line where the profile is applied, otherwise looks good.

jmencak commented:
/cc @Deepthidharwar for any additional feedback as I know she is experimenting with this right now.

skordas commented:
Looks good. Only missing command how to get logs after [creating new tuned](https://github.com/openshift/openshift-docs/pull/16051#discussion_r311894047)

adellape commented:
Thanks, added the command for grabbing logs:

http://file.rdu.redhat.com/~adellape/072919/hugepages/scalability_and_performance/what-huge-pages-do-and-how-they-are-consumed-by-apps.html#configuring-huge-pages_huge-pages

adellape commented:
/cherrypick enterprise-4.1

adellape commented:
/cherrypick enterprise-4.2
